cloud-minecraft/cloud-bukkit/src/main/java/cloud/commandframework/bukkit/parsers/location/LocationArgument.java

```diff
@@ -0,0 +1,317 @@
+        ) {
+            super(
+                    TypeToken.get(Location.class),
+                    name
+            );
+            this.bukkitCommandManager = bukkitCommandManager;
+        }
+
+        @Override
+        public @NonNull CommandArgument<@NonNull C, @NonNull Location> build() {
+            return new LocationArgument<>(
+                    this.bukkitCommandManager,
+                    this.isRequired(),
+                    this.getName(),
+                    this.getDefaultValue(),
+                    this.getSuggestionsProvider(),
+                    new LinkedList<>()
+            );
+        }
+
+    }
+
+    public static final class LocationParser<C> implements ArgumentParser<C, Location> {
+
+        private final LocationCoordinateParser<C> locationCoordinateParser = new LocationCoordinateParser<>();
+        private final BukkitCommandManager<C> bukkitCommandManager;
+
+        /**
+         * Create a new location parser
+         *
+         * @param bukkitCommandManager Bukkit command manager
+         */
+        public LocationParser(
+                final @NonNull BukkitCommandManager<C> bukkitCommandManager
+        ) {
+            this.bukkitCommandManager = bukkitCommandManager;
+        }
+
+        @Override
+        public @NonNull ArgumentParseResult<@NonNull Location> parse(
+                @NonNull final CommandContext<@NonNull C> commandContext, @NonNull final Queue<@NonNull String> inputQueue
+        ) {
+            if (inputQueue.size() < 3) {
+                final StringBuilder input = new StringBuilder();
+                for (int i = 0; i < inputQueue.size(); i++) {
+                    input.append(((LinkedList<String>) inputQueue).get(i));
+                    if ((i + 1) < inputQueue.size()) {
+                        input.append(" ");
+                    }
+                }
+                return ArgumentParseResult.failure(
+                        new LocationParseException(
+                                commandContext,
+                                input.toString()
+                        )
+                );
+            }
+            final LocationCoordinate[] coordinates = new LocationCoordinate[3];
+            for (int i = 0; i < 3; i++) {
+                final ArgumentParseResult<@NonNull LocationCoordinate> coordinate = this.locationCoordinateParser.parse(
+                        commandContext,
+                        inputQueue
+                );
+                if (coordinate.getFailure().isPresent()) {
+                    return ArgumentParseResult.failure(
+                            coordinate.getFailure().get()
+                    );
+                }
+                coordinates[i] = coordinate.getParsedValue().orElseThrow(NullPointerException::new);
+            }
+            final Location originalLocation;
+            final CommandSender mappedSender = this.bukkitCommandManager.getBackwardsCommandSenderMapper()
```

Comment:
`final CommandSender mappedSender = commandContext.get("BukkitCommandSender");` and then we don't need to supply a command manager to the builder. This is added to the contexts by the BukkitCommandPreprocessor.